### PR TITLE
Usergroup twig work2

### PR DESCRIPTION
**Pull request by tabraiz961**
UserGroups.php file extracted HTML to twig templates 

4 twig templates updated in templates/server/user_groups/*.twig

Ref: #14801

### CODE REVIEW

File: libraries/classes/Server/UserGroups.php [hunk 1]
```diff
@@ -42,28 +43,25 @@ public static function getHtmlForListingUsersofAGroup($userGroup)
...
                 }
-                $html_output .= '</tbody>'
-                    . '</table>';
             }
         }
         $GLOBALS['dbi']->freeResult($result);
-        return $html_output;
+        $template = new Template();
+        return $template->render('/server/user_groups/user_listings', [
+            'user_group_msg' => $userGroupMsg,
```
Comment (williamdes) [inline libraries/classes/Server/UserGroups.php]: you can use `{{ 'Users of \'%s\' user group'|trans|format(userGroup) }}` in twig ;)

File: libraries/classes/Server/UserGroups.php [hunk 2]
```diff
@@ -74,96 +72,75 @@ public static function getHtmlForListingUsersofAGroup($userGroup)
...
-        $html_output .= '<a href="' . Url::getFromRoute('/server/user-groups', ['addUserGroup' => 1]) . '">'
-            . Generator::getIcon('b_usradd')
-            . __('Add user group') . '</a>';
-        $html_output .= '</fieldset></div>';
-
-        return $html_output;
+        $template = new Template();
+        return  $template->render('server/user_groups/user_groups', [
+            'action' => $action,
+            'hidden_inputs' => isset($hidden_inputs)?$hidden_inputs:'',
```
Comment (williamdes) [inline libraries/classes/Server/UserGroups.php]:
```suggestion
            'hidden_inputs' => isset($hidden_inputs) ? $hidden_inputs : '',
```

Do not forget to run `composer run phpcbf` before committing

File: libraries/classes/Server/UserGroups.php [hunk 3]
```diff
@@ -224,44 +201,22 @@ public static function delete($userGroup)
...
-                . '</h2>';
-        }
-
-        $html_output .= '<form name="userGroupForm" id="userGroupForm"'
-            . ' action="' . Url::getFromRoute('/server/user-groups') . '" method="post">';
         $urlParams = [];
+
+        $editUserGroup = '';
+        if ($userGroup != null) {
+            $editUserGroup = sprintf(__('Edit user group: \'%s\''), htmlspecialchars($userGroup));
```
Comment (williamdes) [inline libraries/classes/Server/UserGroups.php]: same here for twig

File: libraries/classes/Server/UserGroups.php [hunk 4]
```diff
@@ -288,31 +243,30 @@ public static function getHtmlToEditUserGroup($userGroup = null)
...
-
-        $html_output .= '<fieldset id="fieldset_user_group_rights_footer"'
-            . ' class="tblFooters">';
-        $html_output .= '<input class="btn btn-primary" type="submit" value="' . __('Go') . '">';
-        $html_output .= '</fieldset>';
-
-        return $html_output;
+        $template = new Template();
+        return $template->render('/server/user_groups/edit_user_groups', [
+//            'user_group' => $userGroup,
```
Comment (williamdes) [inline libraries/classes/Server/UserGroups.php]:
```suggestion
```

**Review by williamdes — CHANGES_REQUESTED** [review]
Good work, some minor changes are required

**Comment by williamdes** [comment]
Can you mention the twig issue here so it links ?

File: libraries/classes/Server/UserGroups.php [hunk 5]
```diff
@@ -328,20 +282,20 @@ public static function getHtmlToEditUserGroup($userGroup = null)
...
-                . ' name="' . $level . '_' . $tab . '" value="Y">';
-            $html_output .= '<label for="' . $level . '_' . $tab . '">'
-                . '<code>' . $tabName . '</code>'
-                . '</label>';
-            $html_output .= '</div>';
+            $tabDetail = [];
+            $tabDetail['in_array'] = (in_array($tab, $selected) ? ' checked="checked"' : '');
+            $tabDetail['tab'] = $tab;
+            $tabDetail['tab_name'] = $tabName;
+            $tabDetails[] =$tabDetail;
```
Comment (williamdes) [inline libraries/classes/Server/UserGroups.php]:
```suggestion
            $tabDetails[] = $tabDetail;
```

File: templates/server/user_groups/user_listings.twig [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<h2>{{ 'Users of \'%s\' user group'| trans | format(user_group_special_chars) }}</h2>
```
Comment (williamdes) [inline templates/server/user_groups/user_listings.twig]:
```suggestion
<h2>{{ 'Users of \'%s\' user group'|trans|format(user_group_special_chars) }}</h2>
```

**Review by williamdes — APPROVED** [review]
LGTM

File: libraries/classes/Server/UserGroups.php (hunk 2 above)
Comment (tabraiz961) [reply libraries/classes/Server/UserGroups.php]: phpcbf gave me tons of fixed files however i pushed only the ones i am working on

File: libraries/classes/Server/UserGroups.php (hunk 3 above)
Comment (tabraiz961) [reply libraries/classes/Server/UserGroups.php]: done

**Comment by tabraiz961** [comment]
@williamdes  i dont understand the designer quotes issue given by Travis CI, i fixed it and was reviewed and pushed to QA_5_0 and these changes were not merged in phpmyadmin/master. My pull request should not have conflicted. Correct me please.

File: libraries/classes/Server/UserGroups.php (hunk 2 above)
Comment (williamdes) [reply libraries/classes/Server/UserGroups.php]: quite strange
did you `composer update` before ?

**Comment by williamdes** [comment]
@tabraiz961 can you check the box to allow me to edit your changes please ?

BTW your history does not make sense to me 
image

So I squashed and rebased the changes:
image


If you allow me to push the change I will help you with travis :)

File: libraries/classes/Server/UserGroups.php (hunk 2 above)
Comment (tabraiz961) [reply libraries/classes/Server/UserGroups.php]: > quite strange
> did you `composer update` before ?

no

**Comment by williamdes** [comment]
@tabraiz961 please run 
```
git fetch --all -p
git checkout usergroup_twig_work2
git reset --hard b427cfae84a7b29c7d20843ed6726bf0fd659316
```
You have one test to fix but it is very easy
https://travis-ci.org/phpmyadmin/phpmyadmin/jobs/637575583#L700

**Comment by tabraiz961** [comment]
> @tabraiz961 please run
> 
> ```
> git fetch --all -p
> git checkout usergroup_twig_work2
> git reset --hard b427cfae84a7b29c7d20843ed6726bf0fd659316
> ```
i have run the commands on local. I hope they meant to pull all working remote changes and reset head to a previous commit which is already in sync with remote.

> You have one test to fix but it is very easy
> https://travis-ci.org/phpmyadmin/phpmyadmin/jobs/637575583#L700

This seems to be an issue although i dont fully understand this section but it seems that 
`$dbi->expects($this->exactly(2))
            ->method('fetchAssoc')
            ->willReturnOnConsecutiveCalls(
                [
                    'usergroup' => 'ug',
                    'tab' => 'server_sql',
                    'allowed' => 'Y',
                ],
                false
            );`
in the UserGroupsTest.php, A usergroup(ug) is created with server_sql checkbox checked. So html should also be rendered with the checked="checked" attribute causing the issue. Am i going in the right direction ?

**Comment by williamdes** [comment]
> i have run the commands on local. I hope they meant to pull all working remote changes and reset head to a previous commit which is already in sync with remote.

It has set your local copy to the remote one, discarding all changes that where made locally before the command

> This seems to be an issue although i dont fully understand this section but it seems that
$dbi->expects($this->exactly(2)) ->method('fetchAssoc') ->willReturnOnConsecutiveCalls( [ 'usergroup' => 'ug', 'tab' => 'server_sql', 'allowed' => 'Y', ], false );
in the UserGroupsTest.php, A usergroup(ug) is created with server_sql checkbox checked. So html should also be rendered with the checked="checked" attribute causing the issue. Am i going in the right direction ?

You seem to have a space problem

First line is what the test wants, second is what you generate, maye consider adjusting the twig template
```diff
+ <input type="checkbox" class="checkall" checked="checked" name="server_sql" value="Y">
- <input type="checkbox" class="checkall"  checked="checked" name="server_sql"  value="Y">
```

**Comment by tabraiz961** [comment]
One last thing since the changes are already approved do i need to push the changes for the space problem?

**Comment by williamdes** [comment]
Yes, please fix this so the tests pass

**Review by williamdes — APPROVED** [review]
Tests pass :heavy_check_mark:

File: templates/server/user_groups/user_listings.twig (hunk 6 above)
Comment (boherm) [inline templates/server/user_groups/user_listings.twig]:
```suggestion
<h2>{{ 'Users of \'%s\' user group' | trans | format(user_group_special_chars) }}</h2>
```

**Review by boherm — APPROVED** [review]

**Comment by williamdes** [comment]
Thank you @tabraiz961 for your contribution !
Merged as 66b446a13ea76686c03ea7dc5812cc1287e5688a

**Pull request merged**